### PR TITLE
SPH order 1 - SPMD compatibility

### PR DESCRIPTION
**Pull request by bertrand-altair**
New way tom compute correction order 1 for SPH + SPMD compatibility (old cold is kept in comment)

### CODE REVIEW

File: engine/source/elements/sph/spcompl.F [hunk 1]
```diff
@@ -71,19 +71,20 @@ SUBROUTINE SPCOMPL(
       INTEGER N,INOD,JNOD,J,NVOIS,M,NN,
      .        NVOISS,SM,JS,NC,IC,IS,
      .        I,IPRT,IGEO,IORDER,NMUN,NZERO,NUN,
-     .        MWAMUN(NSPHACT),MWAUN(NSPHACT),MWAZERO(NSPHACT)
+     .        MWAMUN(NSPHACT),MWAUN(NSPHACT),MWAZERO(NSPHACT),
+     .        SPH_IORD1
```
Comment (sebastienVilleneuve) [inline engine/source/elements/sph/spcompl.F:75]: INTENT statement is missing here

File: engine/source/elements/sph/spcompl.F [hunk 2]
```diff
@@ -109,677 +110,840 @@ SUBROUTINE SPCOMPL(
...
+          WGRADYYI=WGRADYYI+VJ*WGRAD(2)*(YI-YJ)
+          WGRADYZI=WGRADYZI+VJ*WGRAD(2)*(ZI-ZJ)
+          WGRADZXI=WGRADZXI+VJ*WGRAD(3)*(XI-XJ)
+          WGRADZYI=WGRADZYI+VJ*WGRAD(3)*(YI-YJ)
+          WGRADZZI=WGRADZZI+VJ*WGRAD(3)*(ZI-ZJ)
+        ENDDO
+C------
+C       AlphaI
+C        ALPHAI=ONE/MAX(EM20,WCOMPI)
+        WACOMP( 2,N)=ONE/SIGN(MAX(EM20,ABS(WCOMPXI)),WCOMPXI)
```
Comment (sebastienVilleneuve) [inline engine/source/elements/sph/spcompl.F:376]: ' ' char here : WACOMP( 2,N)

File: engine/source/mpi/elements/spmd_sph.F [hunk 3]
```diff
@@ -693,6 +693,7 @@ SUBROUTINE SPMD_SPHGETW(SPBUF,WACOMP,WA,WAR)
 C-----------------------------------------------
 C   D u m m y   A r g u m e n t s
 C-----------------------------------------------
+       INTEGER SPH_IORD1
```
Comment (sebastienVilleneuve) [inline engine/source/mpi/elements/spmd_sph.F:696]: INTENT statement is missing

File: engine/source/mpi/elements/spmd_sph.F [hunk 4]
```diff
@@ -1406,6 +1493,7 @@ SUBROUTINE SPMD_SPHGETG(WGRADT,WACOMP,WGR)
...
```
Comment (sebastienVilleneuve) [inline engine/source/mpi/elements/spmd_sph.F:1496]: INTENT statement is missing

File: engine/source/elements/sph/sphprep.F [hunk 5]
```diff
@@ -117,7 +118,7 @@ SUBROUTINE SPHPREP(
      .    IBUFSSG_IO(SIBUFSSG_IO), LGAUGE(3,*),
      .    NGROUNC,  IGROUNC(*), SOL2SPH(2,*), SPH2SOL(*), IXS(NIXS,*),
      .    IADS(8,*), ADDCNE(*), WASPACT(*), ICONTACT(*), OFF_SPH_R2R(*),
-     .    IRUNN_BIS
+     .    IRUNN_BIS,SPH_IORD1
```
Comment (sebastienVilleneuve) [inline engine/source/elements/sph/sphprep.F:121]: INTENT statement missing

File: engine/source/elements/forintp.F [hunk 6]
```diff
@@ -141,7 +141,8 @@ SUBROUTINE FORINTP(
      .   IGEO(NPROPGI,*),
      .   LPRTSPH(2,0:NPART),LONFSPH(*),WASPACT(*),ISPHIO(NISPHIO,*),
      .   ITASK,GRTH(*),IGRTH(*), LGAUGE(3,*), NGROUNC, IGROUNC(*), 
-     .   IXS(NIXS,*), IRST(3,*), SOL2SPH(2,*), SPH2SOL(*), SOL2SPH_TYP(*)
+     .   IXS(NIXS,*), IRST(3,*), SOL2SPH(2,*), SPH2SOL(*), SOL2SPH_TYP(*),
+     .   SPH_IORD1
```
Comment (sebastienVilleneuve) [inline engine/source/elements/forintp.F:145]: INTENT statement missing

File: engine/source/elements/sph/spcompl.F (hunk 1 above)
Comment (bertrand-altair) [reply engine/source/elements/sph/spcompl.F]: done

File: engine/source/elements/sph/spcompl.F (hunk 2 above)
Comment (bertrand-altair) [reply engine/source/elements/sph/spcompl.F]: done

File: engine/source/mpi/elements/spmd_sph.F (hunk 4 above)
Comment (bertrand-altair) [reply engine/source/mpi/elements/spmd_sph.F]: done

File: engine/source/elements/sph/sphprep.F (hunk 5 above)
Comment (bertrand-altair) [reply engine/source/elements/sph/sphprep.F]: done

File: engine/source/elements/forintp.F (hunk 6 above)
Comment (bertrand-altair) [reply engine/source/elements/forintp.F]: done

**Comment by sebastienVilleneuve** [comment]
Hi @bertrand-altair 

In this PR there are 2 commit , could you squash them ?

Thanks,
Sébastien

**Review by sebastienVilleneuve — CHANGES_REQUESTED** [review]
Hi @bertrand-altair 

I can see 2 commits in this PR , can you squash them ?

Thanks,
Sébastien

**Review by sebastienVilleneuve — APPROVED** [review]

**Pull request merged**